source/NVDAHelper.py

```diff
@@ -398,7 +398,14 @@ def nvdaControllerInternal_inputLangChangeNotify(threadID,hkl,layoutString):
 def nvdaControllerInternal_typedCharacterNotify(threadID,ch):
 	focus=api.getFocusObject()
 	if focus.windowClassName!="ConsoleWindowClass":
-		eventHandler.queueEvent("typedCharacter",focus,ch=ch)
+		# Manually queue a call to executeEvent rather than using queueEvent,
+		# As Currently queueEvent uses a lock and there is a small chance it could deadlock
+		# If garbage collection within the lock
+		# caused a COM object from the same thread as the typed character to be released.
+		queueHandler.queueFunction(

```

Comment:
This bypasses the `pendingEvents` mechanism. Perhaps it is worth investigating disabling the garbage collection?